### PR TITLE
f-droid master

### PR DESCRIPTION
**Pull request by flxholle**
Refer to #499

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,10 +8,7 @@
 
 Shuttle is an open source, local music player for Android.
 
-Shuttle comes in two flavours: 
-
-- [Shuttle (free)](https://play.google.com/store/apps/details?id=another.music.player)
-- [Shuttle+](https://play.google.com/store/apps/details?id=com.simplecity.amp_pro)
+# Just download the source code and build your own tracker-free version!
```
Comment (timusus) [inline README.md]: Let's not call it a 'tracker-free' version. Happy with 'analytics free', or some other phrasing .Default Shuttle doesn't contain 'trackers', it contains a library for collecting anonymous crash data and another for collectin anonymous app usage data.

File: app/src/main/java/com/simplecity/amp_library/saf/SafManager.java [hunk 2]
```diff
@@ -221,7 +220,7 @@ private String getExtSdCardFolder(final File file) {
                 }
             }
         } catch (NoSuchMethodError e) {
-            Crashlytics.log("getExtSdCardPaths() failed. " + e.getMessage());
+
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/saf/SafManager.java:223]: We still need to log when exceptions occur. (`Log.e(getExtSdCardPaths() failed. " + e.getMessage())`)

File: app/src/main/java/com/simplecity/amp_library/services/Equalizer.java [hunk 3]
```diff
@@ -304,7 +303,7 @@ public synchronized void update() {
                 updateDsp(mAudioSessions.get(sessionId));
             }
         } catch (NoSuchMethodError e) {
-            Crashlytics.log("No such method error thrown when updating equalizer.. " + e.getMessage());
+
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/services/Equalizer.java:306]: Need to retain a log message (`Log.e(..)`)

File: app/src/main/java/com/simplecity/amp_library/tagger/TaggerUtils.java [hunk 4]
```diff
@@ -164,7 +163,7 @@ static boolean requiresPermission(List<String> paths) {
                 }
             }
         } catch (NoSuchMethodError e) {
-            Crashlytics.log("getExtSdCardPaths() failed. " + e.getMessage());
+
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/tagger/TaggerUtils.java:166]: Need to retain a log message (Log.e(..))

File: app/src/main/java/com/simplecity/amp_library/ui/adapters/LoggingViewModelAdapter.java [hunk 5]
```diff
@@ -23,15 +21,11 @@ public LoggingViewModelAdapter(String tag) {
     @Override
     public synchronized Disposable setItems(List<ViewModel> items, @Nullable CompletionListUpdateCallback callback) {
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/ui/adapters/LoggingViewModelAdapter.java:23]: Need to retain a log message (Log.d(..))

File: app/src/main/java/com/simplecity/amp_library/ui/adapters/LoggingViewModelAdapter.java [hunk 6]
```diff
@@ -23,15 +21,11 @@ public LoggingViewModelAdapter(String tag) {
...
 
-        Crashlytics.log(Log.DEBUG, TAG, String.format("setItems called for: '%s'", tag));
-
         return super.setItems(items, new CompletionListUpdateCallback() {
 
             @Override
             public void onComplete() {
 
-                Crashlytics.log(Log.DEBUG, TAG, String.format("setItems complete for: '%s'. Dispatching updates.", tag));
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/ui/adapters/LoggingViewModelAdapter.java:33]: Need to retain a log message (Log.d(..))

File: app/src/main/java/com/simplecity/amp_library/utils/LogUtils.java [hunk 7]
```diff
@@ -18,8 +17,7 @@ public static void logException(String tag, String message, @Nullable Throwable
                 throwable.printStackTrace();
             }
         } else {
-            Crashlytics.log(Log.ERROR, tag, message + "\nThrowable: " + (throwable != null ? throwable.getMessage() : null));
-            Crashlytics.logException(throwable);
+
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/utils/LogUtils.java:20]: Need to retain a log message (Log.e(..))

File: app/src/main/java/com/simplecity/amp_library/utils/PlaylistUtils.java [hunk 8]
```diff
@@ -466,7 +465,7 @@ public static Playlist createPlaylist(Context context, String name) {
                                 id = Long.parseLong(uri.getLastPathSegment());
                             }
                         } catch (NullPointerException e) {
-                            Crashlytics.log("Failed to create playlist: " + e.getMessage());
+
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/utils/PlaylistUtils.java:468]: Need to retain a log message (Log.e(..))

File: app/src/main/java/com/simplecity/amp_library/utils/PlaylistUtils.java [hunk 9]
```diff
@@ -478,7 +477,6 @@ public static Playlist createPlaylist(Context context, String name) {
         if (id != -1) {
             playlist = new Playlist(Playlist.Type.USER_CREATED, id, name, true, false, true, true, true);
         } else {
-            Crashlytics.log(String.format("Failed to create playlist. Name: %s, id: %d", name, id));
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/utils/PlaylistUtils.java:481]: Need to retain a log message (Log.e(..))

File: app/src/main/java/com/simplecity/amp_library/utils/ThreadUtils.java [hunk 10]
```diff
@@ -15,7 +14,7 @@ public static void ensureNotOnMainThread() {
             if (BuildConfig.DEBUG) {
                 throw new IllegalStateException("ensureNotOnMainThread failed.");
             } else {
-                CrashlyticsCore.getInstance().log("ThreadUtils ensureNotOnMainThread() failed");
+
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/utils/ThreadUtils.java:17]: Need to retain a log message (Log.e(..))

**Review by timusus — CHANGES_REQUESTED** [review]
Just need to make sure we're still logging when exceptions occur, now that Crashlytics.log() has been removed.

File: .idea/codeStyles/Project.xml [hunk 11]
```diff
@@ -1,229 +1,5 @@
 <component name="ProjectCodeStyleConfiguration">
   <code_scheme name="Project" version="173">
```
Comment (timusus) [inline .idea/codeStyles/Project.xml:2]: I'm not sure why we're removing the code style?

**Review by timusus — CHANGES_REQUESTED** [review]
There are a bunch of changes in 83121a9 that I don't really understand.

If these are merely 'suggestions', then I'd rather not incorporate them. They just create a fork from the master branch, which means additional maintenance overhead. I don't know why f-droid cares about the .idea settings.

**Comment by flxholle** [comment]
Well all the changes in the .idea folder were not made by me, I think this are automatic changes. Sorry.
Also as you have mentioned that the Crashlytics.log() should be replaced with Log.e() and the things happend to the .idea folder, I will now create a new "clear" branch and remake my changes

**Comment by timusus** [comment]
No worries, that's OK. You don't _have_ to make a fresh branch - you can probaby just revert 83121a9. And then add in those log statements in a separate commit.

Up to you. I appreciate your work :)

**Comment by flxholle** [comment]
Well second try :sweat_smile: 
#504

**Pull request closed without merging**